### PR TITLE
fix commandWithArgsRegex

### PR DESCRIPTION
**Pull request by zirain**
Commit Message: followup https://github.com/envoyproxy/envoy/pull/34369
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] #33569
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Review by mattklein123 — DISMISSED** [review]
Does this need an additional release note? What's the history here?

/wait

**Comment by zirain** [comment]
> Does this need an additional release note? What's the history here?
> 
> /wait

there's a release notes in https://github.com/envoyproxy/envoy/pull/34369, I found https://github.com/envoyproxy/envoy/pull/34369 had some problem when trying to use it in istio master.

**Comment by zirain** [comment]
@mattklein123 @kyessenov or anyoneelse, who can take a look at this?

File: source/common/formatter/substitution_formatter.cc [hunk 1]
```diff
@@ -38,7 +38,7 @@ const std::regex& SubstitutionFormatParser::commandWithArgsRegex() {
   //                              ____________________  _____________________ _____________
   //                              |                  |  |                   | |           |
   CONSTRUCT_ON_FIRST_USE(std::regex,
-                         R"EOF(^%((?:[A-Z]|[0-9]|_)+)(?:\(([^\)]*|[^\}]*)\))?(?::([0-9]+))?%)EOF");
+                         R"EOF(^%((?:[A-Z]|[0-9]|_)+)(?:\(([^\)]*|[^\%]*)\))?(?::([0-9]+))?%)EOF");
```
[the remark this answers is not in the document]
Comment (kyessenov) [reply source/common/formatter/substitution_formatter.cc:41]: I think the regex simply needs to say to match anything lazily as a subcommand that's not ')%" or '):982%' (for any digits).

So the right regex is simply:
``` R"EOF(^%((?:[A-Z]|[0-9]|_)+)(?:\((.*?)\))?(?::([0-9]+))?%)EOF");```

**Review by kyessenov — CHANGES_REQUESTED** [review]
Please see the comment.

**Review by kyessenov — APPROVED** [review]
This effectively make ")%" as a terminating sequence for any subcommand. That's rare enough to be not a problem and we can add escaping later.

**Comment by zirain** [comment]
@mattklein123 can you take a look again?

**Comment by kyessenov** [comment]
@yanavlasov can you take a look? We need a fix merged because the previous PR makes parsing broken.

**Pull request merged**